### PR TITLE
List syntax update

### PR DESCRIPTION
**Pull request by ddalcino**
This implements the splitting of the `aqt list` command into `aqt list-qt` and `aqt list-tool`, as discussed previously in #306. It does not address the other changes mentioned in #306; I think that's best left to another PR.

#306 mentions retaining the old syntax for the "install", "tool", "doc", and "src" commands to avoid breaking backwards compatibility. I have made no attempt to retain backwards compatibility for `aqt list`, because the feature is so new that I don't think anyone really depends on it yet, and I think it would make the code really messy.

### CODE REVIEW

File: aqt/metadata.py [hunk 1]
```diff
@@ -41,6 +41,21 @@
...
+            "https://python-semanticversion.readthedocs.io/en/latest/reference.html#semantic_version.SimpleSpec\n"
+            "Examples:\n"
+            '* "*": matches everything\n'
+            '* "5": matches every version with major=5\n'
+            '* "5.6": matches every version beginning with 5.6\n'
+            '* "5.*.3": matches versions with major=5 and patch=3\n'
+            '* "<5": matches every version with major<5\n'
+            '* ">5,<6.2": matches versions starting with 6.0 and 6.1\n'
+            '* "<=5": matches major<=5, including all preview/prerelease versions'
+        )
```
Comment (ddalcino) [inline aqt/metadata.py:57]: We haven't discussed this at all, but `semantic_version` also provides NpmSpec as an alternative to SimpleSpec. Their docs at https://python-semanticversion.readthedocs.io/en/latest/reference.html#semantic_version.SimpleSpec mention an extension for PyPI-style specs as well, but I haven't been able to locate it yet.

The reason I mention this is that if we use a spec standard that everyone else already knows about, we won't need this `usage()` method to tell them how to use it. I could be very wrong about this, but I think that more of our users will know about npm specs than will know how to use SimpleSpec. I would rather not try to teach people how to use SimpleSpec if there's an alternative that they already know.

File: tests/test_list.py [hunk 2]
```diff
@@ -25,51 +26,53 @@
...
+def test_versions(
+    init_data, expect_str, expect_fmt, expect_flat, expect_last, expect_bool
+):
+    versions = Versions(init_data)
+    assert str(versions) == expect_str
+    assert format(versions) == expect_fmt
+    assert format(versions, "s") == expect_str
+    assert versions.flattened() == expect_flat
+    assert versions.latest() == expect_last
+    assert bool(versions) == expect_bool
```
Comment (ddalcino) [inline tests/test_list.py:70]: This refactor has nothing to do with `list-qt` and `list-tool`, and probably belongs in a separate PR. I can spin it off if necessary.

File: aqt/metadata.py (hunk 1 above)
Comment (miurahr) [reply aqt/metadata.py:57]: IMO, we can drop description following example.

```
            '* "<5": matches every version with major<5\n'
            '* ">5,<6.2": matches versions starting with 6.0 and 6.1\n'
            '* "<=5": matches major<=5, including all preview/prerelease versions'
```

These are mostly a description of simple spec.

File: tests/test_list.py (hunk 2 above)
Comment (miurahr) [reply tests/test_list.py:71]: It is OK.

File: aqt/metadata.py (hunk 1 above)
Comment (miurahr) [reply aqt/metadata.py:57]: It is no harm but IMHO many users may not use these complex expressions.

Comment (ddalcino) [reply aqt/metadata.py]: Ok, I removed the last 3; if users need more help they can read the docs.

**Pull request merged**